index.js

```diff
@@ -349,13 +354,15 @@ async function fastifyStatic (fastify, opts) {
             continue
           }
           routes.add(route)
-          fastify.head(route, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, '/' + file, rootPath)
-          })
-
-          fastify.get(route, routeOpts, function (req, reply) {
-            pumpSendToReply(req, reply, '/' + file, rootPath)
-          })
+          const matchedRouteOpts = {
+            ...routeOpts,
+            method: ['HEAD', 'GET'],
+            url: route,
+            handler
+          }
+          matchedRouteOpts.config = matchedRouteOpts.config || {}
```

Comment:
you can add `config: {}` before `...routeOpts` to avoid this mutation.

Reply:
pretty much why I did things this way.

Reply:
I usually do the following branchless code. Avoiding the mutation of a variable by first checking it is imho faster than always setting it, even if it is the previous value.

```js
matchedRouteOpts.config || (matchedRoutesOpts.config = {})
```

Reply:
Does it better than `matchedRouteOpts.config ??= {}`?

Reply:
Unfortunately it's not yielding anything in this case.